components/brave_wallet/common/zcash_utils.cc

```diff
@@ -5,18 +5,91 @@
 
 #include "brave/components/brave_wallet/common/zcash_utils.h"
 
-#include <optional>
+#include <string>
+#include <utility>
+#include <vector>
 
+#include "base/big_endian.h"
+#include "base/types/expected.h"
 #include "brave/components/brave_wallet/common/btc_like_serializer_stream.h"
 #include "brave/components/brave_wallet/common/encoding_utils.h"
+#include "brave/components/brave_wallet/common/f4_jumble.h"
 #include "brave/components/brave_wallet/common/hash_utils.h"
 #include "brave/third_party/bitcoin-core/src/src/base58.h"
+#include "brave/third_party/bitcoin-core/src/src/bech32.h"
+#include "brave/third_party/bitcoin-core/src/src/util/strencodings.h"
+#include "components/base32/base32.h"
 
 namespace brave_wallet {
 
 namespace {
 constexpr size_t kPubKeyHashSize = 20;
 constexpr size_t kPrefixSize = 2;
+
+// https://zips.z.cash/zip-0316#encoding-of-unified-addresses
+enum AddrType {
+  P2PKH = 0x00,
+  P2PSH = 0x01,
+  Sapling = 0x02,
+  Orchard = 0x03,
+  End = 0x04
+};
+
```

Comment:
The spec does not mention End = 0x04. What is it?

Reply:
Just to check whether code is known